### PR TITLE
Fixed bug where invalid address message is displayed only for billing address even if both are invalid

### PR DESCRIPTION
**Pull request by LittleBigDev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | At first loading of checkout page, if different addresses are selected for billing and shipping, and both addresses are invalid, a warning message should be displayed for both addresses. But only billing has its warning message displayed.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | [BOOM-4085](http://forge.prestashop.com/browse/BOOM-4085)
| How to test?  | See screen record in ticket

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8435)

### CODE REVIEW

**Comment by eternoendless** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[classes/checkout/CheckoutAddressesStep.php, line 277 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8435#-KxDMOlL61vVu0MC6Gw3:-KxDMOlL61vVu0MC6Gw4:bcv3j79) ([raw file](https://github.com/prestashop/prestashop/blob/7666476c41158ee490567a5988be4bb7d1c779f0/classes/checkout/CheckoutAddressesStep.php#L277)):*
> ```PHP
>         foreach ($errors as $errorKey => $errorMessage) {
>             $params[$errorKey] = $errorMessage;
>         }
> ```

You can use `array_replace($params, $errors)` instead

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8435)*

**Comment by LittleBigDev** [comment]
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[classes/checkout/CheckoutAddressesStep.php, line 277 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8435#-KxDMOlL61vVu0MC6Gw3:-KxDlbxWBIR2a-pCskCJ:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/7666476c41158ee490567a5988be4bb7d1c779f0/classes/checkout/CheckoutAddressesStep.php#L277)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

You can use `array_replace($params, $errors)` instead
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8435)*

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8435#-:-KxDrZ8-POoF7kzVoUN6:bnfp4nl)*

**Review by eternoendless — DISMISSED** [review]

**Comment by marionf** [comment]
Hello @LittleBigDev 
Please see screenrecord24 on the [BOOM-4085](http://forge.prestashop.com/browse/BOOM-4085) ticket.

**Review by toutantic — APPROVED** [review]

**Comment by marionf** [comment]
Hello @LittleBigDev 
I am sorry but I can't reproduce anymore on develop branch the issue described in the [ticket](http://forge.prestashop.com/browse/BOOM-4085) and @vincentbz neither.
So, is this PR is really necessary ?

**Comment by LittleBigDev** [comment]
Yes it is. It's a very specific scenario, few chances to happen in real life. But code definitely needed to be strengthened. It's done.

File: classes/checkout/CheckoutAddressesStep.php [hunk 1]
```diff
@@ -202,51 +214,77 @@ public function handleRequest(array $requestParams = array())
...
             : array();
 
-        if (!empty($checkoutWarning)) {
-            if ($idAddressDelivery === (int)$checkoutWarning['id_address']) {
-                $addressWarning = $checkoutWarning;
-                $errorKey = 'delivery_address_error';
-            } elseif ($idAddressInvoice === (int)$checkoutWarning['id_address']) {
-                $addressWarning = $checkoutWarning;
-                $errorKey = 'invoice_address_error';
-                $this->use_same_address = false;
```
Comment (Quetzacoalt91) [inline classes/checkout/CheckoutAddressesStep.php]: By refactoring this specific line, you made the code less logical. We must not forget that the key 'invoice_address_error' has now consequences in the next test.

Comment (LittleBigDev) [reply classes/checkout/CheckoutAddressesStep.php]: Indeed. We needed both keys (`delivery_address_error` and `invoice_address_error`) to exist at the same time when relevant.

**Pull request merged**